packages/next/src/server/base-server.ts

```diff
@@ -3047,16 +3027,10 @@ export default abstract class Server<
             // distinguishing between `force-static` and pages that have no
             // postponed state.
             // TODO: distinguish `force-static` from pages with no postponed state (static)
-            revalidate: 0,
+            revalidate: isDynamicRSCRequest ? 0 : cacheEntry.revalidate,
           }
         }
 
-        if (typeof cachedData.pageData !== 'string') {
-          throw new Error(
-            `Invariant: expected pageData to be a string, got ${typeof cachedData.pageData}`
-          )
-        }
```

Comment:
To simplify this, the invariant was removed as now we just use either the `pageData` or the `html` from the `cachedData` when responding.